### ISSUE TITLE
org-wild-notifier-affairs buffer replaces ongoing buffer on a timer

### ISSUE THREAD

**Issue by OverkillGuy** [akhramov/org-wild-notifier.el#16]
I'm a very happy user of your library, but a few weeks ago, the buffer `*org-wild-notifier-affairs*` started popping up everywhere, replacing whatever currently active buffer is out with this empty one, long enough to be distracting (2s), and doing so every few minutes.

Visually, it looks like Emacs is possessed, hiding your ongoing work. Annoying as hell.

This seems to be the timer used to check for events, which triggers org-agenda with this new (temporary) buffer name.

Any chance we can make that buffer hidden (background, non-focused?)

I don't know enough of org-agenda to help diagnose there, and the issue mostly seems to be on their side.

If it helps, I started hitting the issue around the time when issue of broken org narrowing was discovered (see https://github.com/akhramov/org-wild-notifier.el/issues/10#issuecomment-412193468 on issue 10).

Thanks in advance!

---

My setup, for reference:

```emacs-lisp
(use-package org-wild-notifier
  :demand t
  :config
  ;; Any even in calendar should ring me up
  (setq org-wild-notifier-keyword-whitelist nil
	;; Use property NOTIFY to specify when to remind me (n minutes before)
	org-wild-notifier-alert-times-property "NOTIFY"
	;; By default, notify X minutes before event
	org-wild-notifier-alert-time 3)
  ;; Toggle alerts on launch
  (org-wild-notifier-mode))
```

**Comment by akhramov** [comment]
Hi, thanks for the report. 
it sounds like you're experiencing a performance problem. Do you experience the same issue on a smaller number of org entries?

Nevertheless, I think it's time to redesign the library using https://github.com/jwiegley/emacs-async.

**Comment by OverkillGuy** [comment]
Hmm, I don't mind the time it takes (especially since any perf issue would be org-mode's problem, no yours).
Instead it's the UI issue of replacing the active buffer with an empty one (the temporary org-agenda one) for any length of time, every 60s.
Will provide a short screencast showing the behaviour.
I'm not against a redesign with async though =p

---
## Reproduction steps

- Run `M-x org-wild-notifier-check` (or wait 60s for timer
- Observe active buffer replaced with the temporary org-agenda one for a brief time

I've been able to `edebug` it to `org-agenda-list` call: https://github.com/akhramov/org-wild-notifier.el/blob/e1ab21134e250ad0694eed4f03530d3ed1899270/org-wild-notifier.el#L268 which pops the new org buffer.

Is there some way to make the org-agenda fetching _not_ pop a new buffer?
I see `org-agenda-sticky` which would enable a (background) buffer to stick around between uses.

**Comment by akhramov** [comment]
@OverkillGuy Thanks for your research, also i'm sorry for the late response.

I've been playing around with org-agenda-sticky and it simply didn't work for me. 
Anyway, the async approach seem to work quite well. Can you please try #17?
Unfortunately, all tests are red so I can't merge / release this version. But it works :)

**Comment by OverkillGuy** [comment]
No biggie, we all have things to do.

I tried it at work (by playing it hardcore and overwriting the .el file MELPA gave me, then restarting emacs/reloading config), and can't reproduce the issue anymore, the buffer never pops in frame!

This is great, thanks for migrating to  async! It also gives me sample code to look at for learning elisp.

**Comment by akhramov** [comment]
Fixed in v0.3.0, please wait for MELPA archives to update.
